### DISCUSSION TITLE
Github Copilot fails to authenticate even though I have access

### DISCUSSION THREAD

**Discussion by tresabhi** [community/community#9404] [Copilot Conversations] [answered]
I have access to Github Copilot under the account [TresAbhi](https://github.com/TresAbhi):
![image](https://user-images.githubusercontent.com/43380238/147711562-19c05c1e-443e-46bf-b073-22d23818ce7e.png)

I am also logged into my VSCode with that account:
![image](https://user-images.githubusercontent.com/43380238/147711605-b31336bc-d069-4a06-9e9c-513bf99d136f.png)

Github Copilot has access to this account:
![image](https://user-images.githubusercontent.com/43380238/147711636-8aa07f4d-0ab9-49e0-b528-27bf3435c1a5.png)

However, Github Copilot fails to function at all. A bit of digging reveals that it fails to authenticate me:
```
[INFO] [default] [2021-12-29T23:54:08.036Z] Experimentation service initialized
[INFO] [default] [2021-12-29T23:54:08.063Z] Stringified features: {}
[INFO] [default] [2021-12-29T23:54:08.063Z] Feature for copilotaa: false
[INFO] [default] [2021-12-29T23:54:08.063Z] Feature for copilotincludeprompt: none
[INFO] [default] [2021-12-29T23:54:08.063Z] Feature for copilotset1: undefined
[INFO] [default] [2021-12-29T23:54:08.063Z] Feature for copilotghgh: false
[INFO] [default] [2021-12-29T23:54:08.063Z] Feature for copilotfim: false
[INFO] [auth] [2021-12-29T23:54:08.640Z] Invalid copilot token: missing token: 404 
[ERROR] [default] [2021-12-29T23:54:08.642Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"
```

Any fixes for this? I have done everything right so far.

**Comment by davecheney** [comment, marked as answer]
> Invalid copilot token: missing token: 404 

Something is wrong with the authentication token that VSCode is using to talk to github. 
<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.

**Reply by TarunavBA** [reply]
This ain't an answer @davecheney, it is not working for me.

**Reply by nobody5050** [reply]
same

**Reply by davecheney** [reply]
@TarunavBA as we have discussed several times, your difficulties are because of your local network. Your computer and/or your local network are preventing your from connecting to the copilot servers. I'm sorry this is the answer, and you are unable to access Copilot, but there isn't much I can do.

**Reply by tresabhi** [reply]
I'll give this a try; untill then, I have removed this as the answer.

**Reply by tresabhi** [reply]
No, this does not fix the issue.

**Reply by davecheney** [reply]
I'm sorry to hear that. The short version is 404 from api.github.com means there was a problem with the OAuth token that vscode used. Sadly there aren't a lot of controls over this process so the best I can recommend, if signing out/in did not work, is to start uninstalling extensions, maybe vscode, deleting preferences, etc.

**Reply by nobody5050** [reply]
interesting, I was really looking forward to using it. Is there a way to see more detailed logs than just the 404 so I can troubleshoot?

**Reply by tresabhi** [reply]
Do note this though: Github copilot **has worked on other machines**, it's just being bugging on this device.

**Reply by davecheney** [reply]
I'm afraid that's all there is. Do you have another computer you could use to try vscode on? Or, I haven't tried this, I think vscode insiders uses its own preferences store, so you could try downloading the -insiders build and trying that which should behave like an independent vscode installation.

**Reply by tresabhi** [reply]
I have tried insiders and the normal one. No luck.

**Comment by Mahdi-Farahani** [comment]
If you are trying GitHub copilot for the first time, this error indicates that you are not part of the beta invite list. What you can do is you can join the waitlist (https://github.com/features/copilot/signup) and wait for your chance to try the beta.

If you have already got the invitation, then there was an error connecting to api.github.com to request a token to use the Copilot service.

Full Discussion: https://github.com/github/feedback/discussions/7707
